File: backend/app/schemas/review.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class ReviewCardOut(BaseModel):
    id: uuid.UUID
    note_id: uuid.UUID
    card_type: str
    question: str
    answer: str
    next_review_at: datetime | None = None
    ease_factor: float
    interval_days: int
    review_count: int
    is_user_edited: bool
    is_flagged: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class ReviewSubmitRequest(BaseModel):
    card_id: uuid.UUID
    quality: int = Field(..., ge=0, le=5)


class ReviewCardUpdate(BaseModel):
    question: str = Field(..., min_length=1)
    answer: str = Field(..., min_length=1)


class ReviewStats(BaseModel):
    total_cards: int
    due_today: int
    mastered_count: int
    learning_count: int
    new_count: int

File: frontend/src/api/review.ts
import client from './client'
import type { ApiResponse } from './notes'

export interface ReviewCard {
  id: string
  note_id: string
  card_type: 'concept' | 'code' | 'scenario'
  question: string
  answer: string
  next_review_at: string | null
  ease_factor: number
  interval_days: number
  review_count: number
  is_user_edited: boolean
  is_flagged: boolean
  created_at: string
}

export interface ReviewStats {
  total_cards: number
  due_today: number
  mastered_count: number
  learning_count: number
  new_count: number
}

export const reviewApi = {
  getToday: () =>
    client.get<any, ApiResponse<ReviewCard[]>>('/review/today'),

  getCards: (noteId: string) =>
    client.get<any, ApiResponse<ReviewCard[]>>(`/review/cards/${noteId}`),

  submitReview: (cardId: string, quality: number) =>
    client.post<any, ApiResponse<ReviewCard>>('/review/submit', { card_id: cardId, quality }),

  getStats: () =>
    client.get<any, ApiResponse<ReviewStats>>('/review/stats'),

  generateCards: (noteId: string) =>
    client.post<any, ApiResponse<ReviewCard[]>>(`/review/generate/${noteId}`),

  editCard: (cardId: string, data: { question: string; answer: string }) =>
    client.put<any, ApiResponse<ReviewCard>>(`/review/cards/${cardId}`, data),

  flagCard: (cardId: string) =>
    client.post<any, ApiResponse<ReviewCard>>(`/review/cards/${cardId}/flag`),
}

